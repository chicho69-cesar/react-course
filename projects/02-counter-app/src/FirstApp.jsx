import PropTypes from 'prop-types'
import { Fragment } from 'react'

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <Fragment>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </Fragment>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string
}
