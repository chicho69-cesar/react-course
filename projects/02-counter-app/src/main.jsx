import './index.css'

import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { CounterApp } from './CounterApp'

// import { HelloWorldApp } from './HelloWorldApp.jsx'
// import { FirstApp } from './FirstApp'

const rootElement = document.getElementById('root')
const root = createRoot(rootElement)

root.render(
  <StrictMode>
    {/* <HelloWorldApp /> */}
    
    {/* <FirstApp
      title={'Hola, soy Goku'}
      subtitle={'Soy un subtÃ­tulo'}
      name={'Goku'}
    /> */}

    <CounterApp value={10} />
  </StrictMode>,
)
