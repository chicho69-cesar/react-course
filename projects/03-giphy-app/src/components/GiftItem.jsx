export const GiftItem = ({ title, url, id }) => {
  return (
    <div className="card">
      <img src={url} alt={`Imagen ${id} - ${title}`} />
      <p>{title}</p>
    </div>
  )
}
