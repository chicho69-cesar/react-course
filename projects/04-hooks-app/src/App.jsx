function App() {
  return (
    <h1>
      Hooks App!!!
    </h1>
  )
}

export default App
