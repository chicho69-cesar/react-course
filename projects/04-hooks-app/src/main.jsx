import './index.css'

// import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'

// import App from './App.jsx'
// import { CounterApp } from './01-use-state/CounterApp'
// import { CounterWithCustomHook } from './01-use-state/CounterWithCustomHook'
// import SimpleForm from './02-use-effect/SimpleForm'
// import { FormWithCustomHook } from './02-use-effect/FormWithCustomHook'
// import { Coords } from './02-use-effect/Coords'
import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { FocusScreen } from './04-use-ref/FocusScreen'

createRoot(document.getElementById('root')).render(
  // <StrictMode>
    // <App />
    // <CounterApp />
    // <CounterWithCustomHook />
    // <SimpleForm />
    // <FormWithCustomHook />
    // <Coords />
    <MultipleCustomHooks />
    // <FocusScreen />
  // </StrictMode>,
)
