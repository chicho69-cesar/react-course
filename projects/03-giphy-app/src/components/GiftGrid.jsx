import { useFetchGifts } from "../hooks/use-fetch-gifts"
import { GiftItem } from "./GiftItem"

export const GiftGrid = ({ category }) => {
  const { gifts, isLoading } = useFetchGifts(category)

  return (
    <>
      <h3>
        {category}
      </h3>

      {isLoading && (
        <h2>Cargando...</h2>
      )}

      <div className="card-grid">
        {gifts.map((gift) => (
          <GiftItem
            key={gift.id}
            {...gift}
          />
        ))}
      </div>
    </>
  )
}
