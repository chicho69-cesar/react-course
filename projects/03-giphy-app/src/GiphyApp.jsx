import { useState } from "react"
import { AddCategory } from "./components/AddCategory"
import { GiftGrid } from "./components/GiftGrid"

function GiphyApp() {
  const [categories, setCategories] = useState(['Dragon Ball'])

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return
    setCategories([newCategory, ...categories])
  }

  return (
    <>
      <h1>
        Giphy App!
      </h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GiftGrid
          key={category}
          category={category}
        />
      ))}
    </>
  )
}

export default GiphyApp
