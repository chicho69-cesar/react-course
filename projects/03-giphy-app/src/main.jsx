import './index.css'

import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'

import GiphyApp from './GiphyApp.jsx'

const rootItem = document.getElementById('root')
const root = createRoot(rootItem)

root.render(
  <StrictMode>
    <GiphyApp />
  </StrictMode>,
)
