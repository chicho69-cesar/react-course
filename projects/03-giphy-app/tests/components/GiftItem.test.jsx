/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import { GiftItem } from '../../src/components'

describe('Test on <GiftItem />', () => {
  const title = 'Saitama'
  const url = 'https://one-punch.com/saitama.jpg'

  test('should match with the snapshot', () => {
    const { container } = render(<GiftItem id={'1'} title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })
  
  test('should show the image with the url and alt', () => {
    render(<GiftItem id={'1'} title={title} url={url} />)

    const { src, alt } = screen.getByRole('img')

    expect(src).toBe(url)
    expect(alt).toEqual(expect.any(String))
  })

  test('should show the title', () => {
    render(<GiftItem id={'1'} title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
