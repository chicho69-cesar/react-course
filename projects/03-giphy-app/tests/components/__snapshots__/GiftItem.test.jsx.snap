// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GiftItem /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Imagen 1 - Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
